db/sqlinterfaces.c

```diff
@@ -3786,18 +3753,12 @@ static void clear_stmt_record(struct sql_state *rec)
  * get_prepared_stmt(), and it is cleaned here as well
  *
  */
-static void put_prepared_stmt(struct sqlthdstate *thd,
-                              struct sqlclntstate *clnt,
-                              struct sql_state *rec, int outrc,
-                              int stored_proc)
+void put_prepared_stmt(struct sqlthdstate *thd, struct sqlclntstate *clnt,
+                       struct sql_state *rec, int outrc)
 {
     sqlite3_stmt *stmt = rec->stmt;
-    if (stmt || ((stored_proc == 1) && (rec->status & CACHE_HAS_HINT))) {
-        if ((gbl_enable_sql_stmt_caching == STMT_CACHE_ALL) ||
-            ((gbl_enable_sql_stmt_caching == STMT_CACHE_PARAM) && clnt->tag)) {
-            if (stmt)
-                sqlite3_reset(stmt);
-
+    if (stmt) {
```

Comment:
do we cache the requests that have no parameters if only STMT_CACHE_PARAM is set with this chance? do we want to do that?

Reply:
Good catch.

I don't like the nesting in `put_prepared_stmt()`. I will simplify (so its atleast easy to read)